src/data-types/date.ts

```diff
@@ -23,14 +23,18 @@ const Date: DataType = {
     const value = parameter.value as any; // Temporary solution. Remove 'any' later.
 
     if (value != null) {
+      const time = new globalDate(value);
+      if ('Invalid Date' === time.toString()) {
+        throw new TypeError('Invalid date.');
+      }
```

Comment:
Can this even happen? The `.validate` method returns a TypeError if an invalid date was passed. 🤔

Reply:
Is the 'validate' method will parse the input value first, then try to validate the date then?

Reply:
I removed the unnecessary checks here and in dateTime as well.